target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -0,0 +1,461 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/* Realtek RTL83XX clock driver
+ *
+ * Copyright (C) 2022 Markus Stockhausen
+ *
+ * This driver provides basic clock support for RTL838X and RTL8389X SoC.
+ * Currently CPU, memory and LXB clock information can be accessed. 
+ * To make use of the driver add the following clocks section to the DTS.
+ *
+ * #include <dt-bindings/clock/rtl83xx-clk.h>
+ * clocks {
+ *   osc: oscillator {
+ *     compatible = "fixed-clock";
+ *     #clock-cells = <0>;
+ *     clock-frequency = <25000000>;
+ *   };
+ *   ccu: clock-controller {
+ *     compatible = "realtek,rtl838x-clock";
+ *     #clock-cells = <1>;
+ *     clocks = <&osc>;
+ *     clock-names = "ref_clk";
+ *   };
+ * };
+ *
+ * Take care to set the compatible value to match your device. E.g.
+ * either realtek,rtl838x-clock or realtek,rtl839x-clock. Afterwards
+ * one can access/identify the clocks in the DTS with <&ccu CLK_CPU>,
+ * <&ccu CLK_MEM> or <&ccu CLK_LXB>.
+ */
+
+#include <linux/clk-provider.h>
+#include <linux/clkdev.h>
+#include <linux/clk.h>
```

Comment:
Heh, my editor sorts these in the opposite order :p

Reply:
sorted